polygeist/tools/cgeist/Lib/CGExpr.cc

```diff
@@ -1189,7 +1189,16 @@ ValueCategory MLIRScanner::VisitDeclRefExpr(DeclRefExpr *E) {
                            /*isReference*/ true);
     }
 
-    auto gv = Glob.GetOrCreateGlobal(VD, /*prefix=*/"");
+    // We need to decide where to put the Global.  If we are in a device
+    // function, the global should be in the gpu module (which is nested inside
+    // another main module).
+    std::pair<mlir::memref::GlobalOp, bool> gv;
+    if (function->getParentOp() != module.get()) {
```

Comment:
`function->getParentOp() != module.get()` doesn't mean the `ParentOp` must be the GPU module, can we check that instead?

Reply:
Done